Sources/Workspace/InitPackage.swift

```diff
@@ -299,6 +284,59 @@ public final class InitPackage {
                         XCTAssertEqual(\(typeName)().text, "Hello, World!")
                     }
                 
+                    static var allTests = [
+                        ("testExample", testExample),
+                    ]
+                }
+
+                """
+        }
+    }
+
+    private func writeExecutableTestsFile(_ path: AbsolutePath) throws {
+        try writePackageFile(path) { stream in
+            stream <<< """
+                import XCTest
+                import class Foundation.Bundle
+
+                final class \(moduleName)Tests: XCTestCase {
+                    func testExample() throws {
+                        // This is an example of a functional test case.
+                        // Use XCTAssert and related functions to verify your tests produce the correct
+                        // results.
+
+                        guard #available(macOS 10.13, *) else {
+                            return
+                        }
+
+                        let fooBinary = productsDirectory.appendingPathComponent("foo")
```

Comment:
I explicitly don't want to do that because then its wrapping `NSProcess` and it would be difficult to create a correct wrapper while keeping the sample code small.